nilearn/plotting/html_stat_map.py

```diff
@@ -154,11 +154,12 @@ def _save_cm(output_cmap, cmap, format='png', n_colors=256):
     imsave(output_cmap, data, cmap=cmap, format=format)
 
 
-def _sanitize_cm(cmap):
+def _sanitize_cm(cmap, n_colors=256):
     """Make sure that the colormap has no duplicated colors.
     """
     # Extract a list of colors
     cmap = plt.cm.get_cmap(cmap)
+    cmap.N = n_colors
```

Comment:
Does the "N" attribute need to be modified on the cmap object, or could we be using n_colors below in the code?

If it's the latter, I would prefer this choice. I never like modifying objects: I worry about side effects.

Reply:
OK, thanks for the advice. I have changed the code to avoid touching the colormap object.